core/src/smartcontracts/wasm.rs

```diff
@@ -295,10 +275,16 @@ impl<'wrld> Runtime<'wrld> {
         let alloc_fn = Self::get_alloc_fn(&mut caller)?;
         let memory = Self::get_memory(&mut caller)?;
 
-        let query = Self::decode_from_memory(&memory, &caller, offset, len)?;
+        let query: QueryBox = Self::decode_from_memory(&memory, &caller, offset, len)?;
 
-        Validator::validate_query(&caller.data().account_id, &query, caller.data().wsv)
-            .map_err(|error| Trap::new(error.to_string()))?;
+        let wsv = caller.data().wsv;
+        wsv.validator_view()
```

Comment:
shouldn't we check the limits on queries as well?

Reply:
I don't know actually. To me it looks strange to even have this checks, but it was introduced before me.

@appetrosyan ?

Reply:
the only reason it exists is that it's direct equivalent of `TransactionLimits` in `transaction.rs`. I think we should limit a query evaluation both for `torii` and in a smart contract otherwise we're vulnerable to DOS. I don't think we do this, but I'm not sure. If this is the case then this requires a separate issue

Reply:
created #3347